src/zkevm_specs/evm/execution/codecopy.py

```diff
@@ -0,0 +1,51 @@
+from ...util import N_BYTES_MEMORY_ADDRESS, FQ
+from ..execution_state import ExecutionState
+from ..instruction import Instruction, Transition
+from ..step import CopyCodeToMemoryAuxData
+from ..table import RW, RWTableTag, CallContextFieldTag, AccountFieldTag
+
+
+def codecopy(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+
+    memory_offset_word, code_offset_word, size_word = (
+        instruction.stack_pop(),
+        instruction.stack_pop(),
+        instruction.stack_pop(),
+    )
+
+    memory_offset, size = instruction.memory_offset_and_length(memory_offset_word, size_word)
+    code_offset = instruction.rlc_to_fq_exact(code_offset_word, N_BYTES_MEMORY_ADDRESS)
+
+    account = instruction.call_context_lookup(CallContextFieldTag.CalleeAddress)
+    code_hash = instruction.account_read(account, AccountFieldTag.CodeHash)
+    code_size = instruction.bytecode_length(code_hash)
+
+    next_memory_size, memory_expansion_gas_cost = instruction.memory_expansion_dynamic_length(
+        memory_offset, size
+    )
+    gas_cost = instruction.memory_copier_gas_cost(size, memory_expansion_gas_cost)
+
+    if not instruction.is_zero(size):
```

Comment:
I missed this in `calldatacopy.py` in #132, please help me also fix that, thanks.
```suggestion
    if instruction.is_zero(size) == FQ(0):
```

Reply:
You mean:
```suggestion
    if instruction.is_zero(size) != FQ(0):
```
right? :)

Reply:
But we want this branch to run when `size != 0`, right?  When `size != 0`, we have `instruction.is_zero(size) == FQ(0)`. The original code `not instruction.is_zero(size)` will always be `False` because `FQ` doesn't implement `__bool__`.